lib/matplotlib/__init__.py

```diff
@@ -424,138 +422,157 @@ def wrapper(*args, **kwargs):
+
+
+@cbook.deprecated("3.0")
 def checkdep_dvipng():
-    try:
-        s = subprocess.Popen([str('dvipng'), '-version'],
-                             stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE)
-        stdout, stderr = s.communicate()
-        line = stdout.decode('ascii').split('\n')[1]
-        v = line.split()[-1]
-        return v
-    except (IndexError, ValueError, OSError):
-        return None
+    return str(get_executable_info("dvipng").version)
 
 
+@cbook.deprecated("3.0")
 def checkdep_ghostscript():
-    if checkdep_ghostscript.executable is None:
-        if sys.platform == 'win32':
-            # mgs is the name in miktex
-            gs_execs = ['gswin32c', 'gswin64c', 'mgs', 'gs']
-        else:
-            gs_execs = ['gs']
-        for gs_exec in gs_execs:
-            try:
-                s = subprocess.Popen(
-                    [gs_exec, '--version'], stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE)
-                stdout, stderr = s.communicate()
-                if s.returncode == 0:
-                    v = stdout[:-1].decode('ascii')
-                    checkdep_ghostscript.executable = gs_exec
-                    checkdep_ghostscript.version = v
-            except (IndexError, ValueError, OSError):
-                pass
+    info = get_executable_info("gs")
+    checkdep_ghostscript.executable = info.executable
+    checkdep_ghostscript.version = str(info.version)
     return checkdep_ghostscript.executable, checkdep_ghostscript.version
 checkdep_ghostscript.executable = None
 checkdep_ghostscript.version = None
 
 
+@cbook.deprecated("3.0")
 def checkdep_pdftops():
-    try:
-        s = subprocess.Popen([str('pdftops'), '-v'], stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE)
-        stdout, stderr = s.communicate()
-        lines = stderr.decode('ascii').split('\n')
-        for line in lines:
-            if 'version' in line:
-                v = line.split()[-1]
-        return v
-    except (IndexError, ValueError, UnboundLocalError, OSError):
-        return None
+    return str(get_executable_info("pdftops").version)
 
 
+@cbook.deprecated("3.0")
 def checkdep_inkscape():
-    if checkdep_inkscape.version is None:
-        try:
-            s = subprocess.Popen([str('inkscape'), '-V'],
-                                 stdout=subprocess.PIPE,
-                                 stderr=subprocess.PIPE)
-            stdout, stderr = s.communicate()
-            lines = stdout.decode('ascii').split('\n')
-            for line in lines:
-                if 'Inkscape' in line:
-                    v = line.split()[1]
-                    break
-            checkdep_inkscape.version = v
-        except (IndexError, ValueError, UnboundLocalError, OSError):
-            pass
+    checkdep_inkscape.version = str(get_executable_info("inkscape").version)
     return checkdep_inkscape.version
 checkdep_inkscape.version = None
```

Comment:
That's just a caching mechanism, which is obsolete because get_executable_info() ls lru_cached.

Reply:
I know, but it doesn't cost me anything to keep it around until checkdep_inkscape itself gets removed.